src/gui_common/GUICommon.cs

```diff
@@ -103,11 +103,13 @@ public void ModulateFadeOut(Control control, float duration)
     /// </summary>
     public TextureRect CreateCompoundIcon(string compoundName, float sizeX = 20.0f, float sizeY = 20.0f)
     {
-        var element = new TextureRect();
-        element.Expand = true;
-        element.RectMinSize = new Vector2(sizeX, sizeY);
-
-        element.Texture = GetCompoundIcon(compoundName);
+        var element = new TextureRect
+        {
+            Expand = true,
+            RectMinSize = new Vector2(sizeX, sizeY),
+            SizeFlagsVertical = (int)Control.SizeFlags.ShrinkCenter,
```

Comment:
I think this doesn't work exactly right as I always see glucose icon having the bottom cutoff, and in weird aspect ratios other icons also get cutoff.
I think I saw a rect size flag for "shrink to fit center with aspect ratio kept" I think that should work the best (unless this is already that).

Reply:
The icon getting cutoff seems like a separate issue that I'm unsure how to fix, it's probably a problem with the texture resolution or something, might worth checking out if that's the case. And yeah, setting the stretch mode to keep aspect should work perfectly, forgot it exists, I'll add that.